src/LootTable.h

```diff
@@ -0,0 +1,620 @@
+	};
+
+
+
+
+
+	/** Any available pool entry type as of 1.16.2 */
+	enum class ePoolEntryType
+	{
+		Item,          // Plain item
+		Tag,           // Descriptor for multiple items. E.g Music Discs, newer arrows...
+		LootTable,     // Another loot table
+		Group,         // Child entries
+		Alternatives,  // Select one entry from the list
+		Sequence,      // Select entries until one cannot be granted
+		Dynamic,       // "generate block specific drops": drops the block itself as item
+		Empty
+	};
+
+
+
+
+	/** Any available types of chest loots as of 1.16.2 */
+	enum class eChestType
+	{
+		AbandonedMineshaft,
+		BuriedTreasure,
+		DesertPyramid,
+		EndCityTreasure,
+		IglooChest,
+		JungleTemple,
+		JungleTempleDispenser,
+		NetherBridge,
+		PillagerOutpost,
+		ShipwreckMap,
+		ShipwreckSupply,
+		ShipwreckTreasure,
+		SimpleDungeon,
+		SpawnBonusChest,
+		StrongholdCorridor,
+		StrongholdCrossing,
+		StrongholdLibrary,
+		UnderwaterRuinBig,
+		UnderwaterRuinSmall,
+
+		/* Village chest types */
+		VillageArmorer,
+		VillageButcher,
+		VillageCartographer,
+		VillageDesertHouse,
+		VillageFisher,
+		VillageFletcher,
+		VillageMason,
+		VillagePlainsHouse,
+		VillageSavannaHouse,
+		VillageShepherd,
+		VillageSnowyHouse,
+		VillageTaigaHouse,
+		VillageTannery,
+		VillageTemple,
+		VillageToolsmith,
+		VillageWeaponsmith,
+
+		WoodlandMansion,
+		None
+	};
+
+	// Declaration of methods in the cpp
+	/** Gets the eType from String. Defaults to Generic */
+	extern enum eType eType(const AString & a_Type);
+	/** Gets the eChestType from String as of 1.16.2 */
+	extern enum eChestType eChestType(const AString & a_Type);
+	extern AString NamespaceConverter(AString a_String);
+	/** Reads keys min and max and writes them into a_Min and a_Max
+	if there are no appropriately named keys there is no change */
+	extern void MinMaxRange(const Json::Value & a_Value, int & a_Min, int & a_Max);
+	/** Reads keys min and max and returns a random integer value in between.
+	a_Noise is used as the random generator and a_Pos is used as the seed */
+	extern int MinMaxRand(const Json::Value & a_Value, const cNoise & a_Noise, const Vector3i & a_Pos, const int & a_Modifier);
+}
+
+
+
+
+
+struct cLootTableCondition;
+typedef std::vector<cLootTableCondition> cLootTableConditionVector;
+/** Represents a condition for a pool item */
+struct cLootTableCondition
+{
+	cLootTableCondition()
+	{
+		m_Type = LootTable::eConditionType::None;
+	}
+
+
+
+
+
+	cLootTableCondition(
+		enum LootTable::eConditionType a_Type,
+		cLootTableConditionVector a_Parameter
+	):
+		m_Type(a_Type),
+		m_Parameter(std::move(a_Parameter))
+	{
+	}
+
+
+
+
+
+	cLootTableCondition(
+		enum LootTable::eConditionType a_Type,
+		Json::Value a_Parameter
+		):
+		m_Type(a_Type),
+		m_Parameter(std::move(a_Parameter))
+	{
+	}
+
+	enum LootTable::eConditionType m_Type;
+	std::variant<cLootTableConditionVector, Json::Value> m_Parameter;
```

Comment:
You don't need to use JSON here. The best way to use a variant is to have each variant represent the type, instead of having a separate tag variable `m_Type`. Just define a struct for each type of condition e.g.

```
/// Checks for a current weather state
struct sWeatherCheck {
    bool m_Raining;  /// If true, the condition evaluates to true only if it's raining.
    bool m_Thundering;  /// If true, the condition evaluates to true only if it's thundering.
};

std::variant<sWeatherCheck, /* etc... */> m_Condition;
```

Then wherever you were switching over `m_Type`, replace it with `std::visit`:
```cpp
std::visit(
	[](const sWeatherCheck& a_Cond)
	{
		bool Res = true;
		if (a_Cond.m_RainCheck)
		{
			Res = Res && (Weather == eWeather_Rain);
		}
		if (a_Cond.m_ThunderCheck)
		{
			Res = Res && (Weather == eWeather_ThunderStorm);
		}
		return Res;
	},
	// etc... for other condition types
    m_Condition);
```

Doing it this way, you should never need to use `std::get`.

Reply:
I thought about creating a struct for every function and condition. 

That's also a nice tool - I have to do some tinkering to make it work.
I haven't tried this but this:
```C
std::visit(
	[](const sWeatherCheck& a_Cond)
	{
		a_Cond(MyArguments);  // The () operator evaluates if the condition
	},
	// etc... for other condition types
    m_Condition);
```
should work.

Some conditions share common arguments or include other conditions

Reply:
this took way to long - but it works now 👍